### PR TITLE
Ol build cleanup

### PR DESCRIPTION
**Pull request by will-moore**
This PR removes all the build steps required by the google closure build of ol3-viewer and other cleanup https://trello.com/c/vyZSm0fj/15-openlayers-5-build-cleanup

Removed:

 - build.xml targets:
    - clean
    - init
    - compile
    - compile-debug
    - copy-to-lib
    - prepare-unit-tests
    - plugin (replaced with 'npm run plugin')
    - build-all
    - build-debug
    - build
- closure/calcdeps.py
- closure/compiler.jar
- ol3_viewer_prepare.sh (deleted in #218)

To test:

 - We should still be able to run the same commands:
    - ```npm run debug```
    - ```npm run prod```
    - ```npm run dev```
    - ```npm run docs``` - This includes ol3-viewer docs in /build/docs/iviewer now (no separate ol3-viewer docs)
    - ```npm run plugin``` which builds viewer plugin only (replaces ```ant build-plugin```)
        - Ideally try to ```omero config append omero.web.apps '"ol3-viewer"'``` and open ```/ol3-viewer/ID``` (see docs below)

 - Check documentation updates:
     - https://github.com/will-moore/omero-iviewer/blob/ol_build_cleanup/README.rst
     - https://github.com/will-moore/omero-iviewer/tree/ol_build_cleanup/plugin/ol3-viewer

Not yet done: ```ant unit-tests```. This will require some more discussion and bigger changes (currently investigating) so would be best to leave to another PR.

### CODE REVIEW

**Comment by will-moore** [comment]
Attempting to get the unit tests working...
I tried running the tests from before #218.

Via ```ant unit-tests```:
It fails with:
```
Error during loading "/Users/wmoore/Desktop/IVIEWER/omero-iviewer/node_modules/karma-babel-preprocessor" plugin:
```


<details><summary>Expand to see the full exception:</summary>

```
unit-tests-only:
     [echo] Fetching test libraries ....
     [exec] npm WARN deprecated circular-json@0.5.9: CircularJSON is in maintenance only, flatted is its successor.
     [exec] 
     [exec] > fsevents@1.2.7 install /Users/wmoore/Desktop/IVIEWER/omero-iviewer/node_modules/karma/node_modules/fsevents
     [exec] > node install
     [exec] 
     [exec] node-pre-gyp WARN Using needle for node-pre-gyp https download 
     [exec] [fsevents] Success: "/Users/wmoore/Desktop/IVIEWER/omero-iviewer/node_modules/karma/node_modules/fsevents/lib/binding/Release/node-v64-darwin-x64/fse.node" is installed via remote
     [exec] + karma-mocha@1.3.0
     [exec] + karma-chrome-launcher@2.2.0
     [exec] + mocha@5.2.0
     [exec] + chai@4.2.0
     [exec] + karma@3.1.4
     [exec] + babel-core@6.26.0
     [exec] + karma-babel-preprocessor@8.0.0
     [exec] + karma-chai@0.1.0
     [exec] + karma-spec-reporter@0.0.32
     [exec] added 266 packages from 964 contributors, removed 2 packages, updated 1 package, moved 2 packages and audited 8130 packages in 14.895s
     [exec] found 42 vulnerabilities (10 low, 23 moderate, 8 high, 1 critical)
     [exec]   run `npm audit fix` to fix them, or `npm audit` for details
     [exec] npm WARN karma-babel-preprocessor@8.0.0 requires a peer of @babel/core@7 but none is installed. You must install peer dependencies yourself.
     [exec] npm WARN @0.0.0 No repository field.
     [exec] npm WARN @0.0.0 No license field.
     [exec] 
     [echo] Running Test Suite ....
     [exec] 22 01 2019 07:45:37.800:ERROR [plugin]: Error during loading "/Users/wmoore/Desktop/IVIEWER/omero-iviewer/node_modules/karma-babel-preprocessor" plugin:
     [exec]   Cannot find module '@babel/core'
     [exec] 22 01 2019 07:45:37.842:ERROR [preprocess]: Can not load "babel", it is not registered!
     [exec]   Perhaps you are missing some plugin?
     [exec] 22 01 2019 07:45:37.957:INFO [karma-server]: Karma v3.1.4 server started at http://0.0.0.0:9876/
     [exec] 22 01 2019 07:45:37.957:INFO [launcher]: Launching browsers ChromeHeadlessNoSandbox with concurrency unlimited
     [exec] 22 01 2019 07:45:37.958:ERROR [karma-server]: Error: Found 2 load errors
     [exec]     at Server.webServer.listen (/Users/wmoore/Desktop/IVIEWER/omero-iviewer/node_modules/karma/lib/server.js:182:27)
     [exec]     at Object.onceWrapper (events.js:273:13)
     [exec]     at Server.emit (events.js:187:15)
     [exec]     at emitListeningNT (net.js:1324:10)
     [exec]     at process._tickCallback (internal/process/next_tick.js:63:19)
     [exec] Result: 1

BUILD FAILED
/Users/wmoore/Desktop/IVIEWER/omero-iviewer/build.xml:257: condition satisfied
```
</details>

I wondered if this was just my local setup, so I tried via ```omero-test-infra```...
```
$ git clone git://github.com/openmicroscopy/omero-test-infra .omero
$ .omero/docker app
...
npm ERR! enoent ENOENT: no such file or directory, open '/tmp/pip-Xina7g-build/package.json'
error: command 'npm' failed with exit status 254
```

<details><summary>Expand to see the full exception:</summary>

```
copying omero_iviewer.egg-info/top_level.txt -> omero-iviewer-0.6.0/omero_iviewer.egg-info
Writing omero-iviewer-0.6.0/setup.cfg
creating dist
Creating tar archive
removing 'omero-iviewer-0.6.0' (and everything under it)
+ pip install dist/omero-iviewer-0.6.0.tar.gz
Processing ./dist/omero-iviewer-0.6.0.tar.gz
Installing collected packages: omero-iviewer
  Running setup.py install for omero-iviewer: started
    Running setup.py install for omero-iviewer: finished with status 'error'
    Complete output from command /usr/bin/python2 -u -c "import setuptools, tokenize;__file__='/tmp/pip-Xina7g-build/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-OHkTJL-record/install-record.txt --single-version-externally-managed --compile:
    running install
    running run_prod
    npm run prod
    npm ERR! Linux 4.9.41-moby
    npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "run" "prod"
    npm ERR! node v6.14.3
    npm ERR! npm  v3.10.10
    npm ERR! path /tmp/pip-Xina7g-build/package.json
    npm ERR! code ENOENT
    npm ERR! errno -2
    npm ERR! syscall open
    
    npm ERR! enoent ENOENT: no such file or directory, open '/tmp/pip-Xina7g-build/package.json'
    npm ERR! enoent ENOENT: no such file or directory, open '/tmp/pip-Xina7g-build/package.json'
    npm ERR! enoent This is most likely not a problem with npm itself
    npm ERR! enoent and is related to npm not being able to find a file.
    npm ERR! enoent
    
    npm ERR! Please include the following file with any support request:
    npm ERR!     /tmp/pip-Xina7g-build/npm-debug.log
    error: command 'npm' failed with exit status 254
    
    ----------------------------------------
Command "/usr/bin/python2 -u -c "import setuptools, tokenize;__file__='/tmp/pip-Xina7g-build/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-OHkTJL-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-Xina7g-build/
You are using pip version 8.1.2, however version 18.1 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```
</details>

So I'm not sure if this was working before all the OpenLayers-5 changes?

**Comment by jburel** [comment]
README:

* ``Intructions on how to add`` => ``Instructions``

File: README.rst [hunk 1]
```diff
@@ -34,35 +35,31 @@ To build an uglified version, run:
...
-Should you like or need to rebuild OMERO.iviewer's internal ol3 viewer,
-please read the section *ol3-viewer* below!
-
 Development
 ===========
 
-Building the uncompressed version of iviewer as described above the application
-can be deployed locally or to a test server, to then be debugged.
-It is also possible to run/debug it with webpack dev-server:
+It is recommended to use the webpack dev-server to build and serve iviewer
```
Comment (jburel) [inline README.rst]: OMERO.iviewer

File: README.rst [hunk 2]
```diff
@@ -34,35 +35,31 @@ To build an uglified version, run:
...
+as this will watch for file saves and re-compile.
```
Comment (jburel) [inline README.rst]: saves?

File: README.rst [hunk 3]
```diff
@@ -125,25 +122,16 @@ To build the html in build/docs, run:
...
 
-
 ol3-viewer
 ==========
 
-The OMERO.iviewer's internal image viewer is based on `OpenLayers 3 <https://openlayers.org/>`_,
+The OMERO.iviewer's internal image viewer is based on `OpenLayers <https://openlayers.org/>`_,
 
-The following software has to be installed in order to compile the java script code:
+For details on how to run and test this viewer independently of the iviewer,
```
Comment (jburel) [inline README.rst]: OMERO.iviewer

File: README.rst [hunk 4]
```diff
@@ -34,35 +35,31 @@ To build an uglified version, run:
...
-- an OMERO server installation
-- an iviewer 'deploy' to that OMERO server:
+You will also need an OMERO.web install with ``omero_iviewer`` installed.
+To add your project to your local OMERO.web install, add the project
+to your ``PYTHONPATH`` and add to ``omero.web.apps``
 
 ::
 
-    $ export PYTHONPATH=$PYTHONPATH:/path/to/iviewer-project-root/plugin
+    $ export PYTHONPATH=$PYTHONPATH:/path/to/omero_iviewer/plugin
```
Comment (jburel) [inline README.rst]: ``omero-iviewer/plugin``

**Comment by jburel** [comment]
Ran ``npm run plugin`` then went to ``http://localhost:4080/ol3-viewer/1``
There is no image displayed

**Comment by jburel** [comment]
Retested ``npm run plugin``.
Switching constantly between servers, I might have missed something in a previous setup
The image is correctly displayed. See screenshot
ol3-viewer


@will-moore I pushed to small commits to the PR, if you are happy with that, I suggest that we merge this PR

**Comment by will-moore** [comment]
Looks good, thanks.

**Pull request merged**